### ISSUE TITLE
cannot log action info

### ISSUE THREAD

**Issue by WillCoco** [storybooks/storybook#2324]
storybook: v3.2.16
react-native: v0.45.1

error info: Attempting to change value of a readonly property
caused by this line: Object.defineProperty(handler, { value: {'name': fnName }});

**Comment by Hypnosphi** [comment]
Seems to work OK with `react-native@0.49.3` (`examples/crna-kitchen-sink`)

@danielduan what can I do to degrade the app to RN 0.45 locally so that I could reproduce this?

**Comment by danielduan** [comment]
In the CRNA example, change the `App.json` expo version. Downgrade the `react`, `react-native`, and `expo` dependencies. @Hypnosphi 

https://github.com/react-community/create-react-native-app/blob/master/VERSIONS.md

**Comment by Hypnosphi** [comment]
And replace `index.js` with `index.android.js`, right?

**Comment by danielduan** [comment]
Oh yeah! It might not matter though because its set up to look for the version default.

**Comment by shilman** [comment]
I'm unable to reproduce on first attempt. Here's what I did:

First, edit `examples/react-native-vanilla/package.json`:
```
    "react": "16.0.0-alpha.12",
    "react-native": "0.45.1"
```

Then:
```
yarn bootstrap --reactnative
cd examples/react-native-vanilla
yarn storybook
yarn react-native run-ios
```

Clicking on the button logs to the actions panel successfully?

**Comment by Tarabyte** [comment]
Same here
 "@storybook/react-native": "^3.2.16",
 "@storybook/addon-actions": "^3.2.16",
 "react": "16.0.0",
 "react-native": "^0.49.3",
 "expo": "^22.0.0",

when running on **android** emulator.

Though works ok when trying to debug in Chrome.

**Comment by Tarabyte** [comment]
@Hypnosphi  You might want to check
```javascript
if (Object.getOwnPropertyDescriptor(handler, 'name').configurable) 
```
in current engine here https://github.com/storybooks/storybook/blob/5cd3dde2ba3ec12a94a9439c1d3a5822dd01ccd4/addons/actions/src/preview.js#L28

And fallback to `eval` version. Or atleast wrap this line with try-catch as a quickfix 😄

**Comment by Hypnosphi** [comment]
I’d prefer to figure out first, why it becomes non-configurable

**Comment by Tarabyte** [comment]
I'd say this is js engine implementation bug. `function.name` indeed is not configurable on android.
Check the note here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name saying it was `configurable: false` before ES2015.

**Comment by hushicai** [comment]
I meet the same issue  too.

but when i `Debug on Remotely`, everything is ok.

react-native: 0.47.2
react-native-storybook:3.2.16

**Comment by Hypnosphi** [comment]
Oh I see. I‘d prefer to keep the name as is rather than use the eval hack in those cases. Would you like to submit a PR?

**Comment by Hypnosphi** [comment]
We could also try to make `handler` truely anonymous before assigning the dynamic name:
```js
  const handler = (() => (..._args) => {
    const args = Array.from(_args).map(_format);
    const channel = addons.getChannel();
    const id = uuid();
    channel.emit(EVENT_ID, {
      id,
      data: { name, args },
    });
  })();
```

**Comment by danielduan** [comment]
Published in 3.2.17
